src/graphapi/opgraph.cpp

```diff
@@ -349,32 +341,36 @@ void BackendOperationGraphDescriptor::setAttribute(miopenBackendAttributeName_t
             std::vector<OpNode*> nodes;
             nodes.reserve(elementCount);
 
-            std::for_each(static_cast<miopenBackendDescriptor_t*>(arrayOfElements),
-                          static_cast<miopenBackendDescriptor_t*>(arrayOfElements) + elementCount,
-                          [&descriptors, &nodes](miopenBackendDescriptor_t apiDescriptor) {
-                              BackendDescriptor& backendDescriptor = deref(apiDescriptor);
-                              if(backendDescriptor.isFinalized())
-                              {
-                                  descriptors.push_back(apiDescriptor);
-                                  nodes.push_back(backendDescriptor.getOperation());
-                              }
-                              else
-                              {
-                                  MIOPEN_THROW(miopenStatusBadParm);
-                              }
-                          });
+            std::for_each_n(static_cast<miopenBackendDescriptor_t*>(arrayOfElements),
+                            elementCount,
+                            [&descriptors, &nodes](miopenBackendDescriptor_t apiDescriptor) {
+                                BackendDescriptor& backendDescriptor = deref(apiDescriptor);
+                                if(backendDescriptor.isFinalized())
+                                {
+                                    descriptors.push_back(apiDescriptor);
+                                    nodes.push_back(backendDescriptor.getOperation());
+                                }
+                                else
+                                {
+                                    MIOPEN_THROW(miopenStatusBadParm, "descriptor not finalized");
+                                }
+                            });
 
             if(!internal::noRepetitions(nodes))
             {
-                MIOPEN_THROW(miopenStatusBadParm);
+                for(size_t i = 0; i < nodes.size(); ++i)
+                {
+                    std::cout << "node: " << nodes[i] << ", desc: " << descriptors[i] << std::endl;
```

Comment:
[R] Use `std::cerr`. Maybe put this under `#ifndef NDEBUG`. Check if it works.

Reply:
Forgot to remove these prints. Will remove.